Next-formalizationFull.hs

```diff
@@ -186,10 +229,65 @@ afterClosed (d, m, y) func int evt = not (beforeOpen (d, m, y) func int evt)
 inperiod:: Date -> (EventInstance -> TimeStamps) -> Int -> Date -> (EventInstance -> TimeStamps) -> Int -> EventInstance -> Bool
 inperiod t1 func1 int1 t2 func2 int2 evt = afterClosed t1 func1 int1 evt && beforeClosed t2 func2 int2 evt
 
+-- is done
+isdone:: EventInstance -> Bool
+isdone e = if List.genericLength (getActualDate e) == 3 then True else False
```

Comment:
I'm looking for a more clean declarative meaninfull definition like

type Date = (Integer, Integer, Integer)
data Timing = Timing {begin :: Date, end :: Date} deriving (Eq, Show)

data EventInstance = EventInstance {name :: String, planned :: Timing, actual :: Timing} deriving (Eq, Show)

event1 = EventInstance{name = "My first event!", planned = Timing{begin = (5, 1, 2017), end = (10, 2, 2017)}, actual = Timing {begin = (6, 1, 2017), end = (16, 3, 2017)}}

showme = begin (planned event1)